### PR TITLE
Refactor code.org top navigation

### PR DESCRIPTION
**Pull request by kelbyhawn**
Refactors the top nav on http://code.org to use modern HTML/CSS to improve accessibility and fix visual bugs:
- Moves the navigation out of the hero banner wrapper on the homepage to avoid overflow issues
  - Required some spacing updates to the homepage hero banners
- Fully responsive and hides the list of links and "Create" button on tablet and mobile 
- Works with LTR and RTL languages
- Works with screen readers and keyboard navigation

_Note:_ I only refactored the `header.haml` view and not any of the dropdowns or hamburger menus.

**Jira ticket:** [ACQ-1369](https://codedotorg.atlassian.net/browse/ACQ-1369)

----

## Desktop

https://github.com/code-dot-org/code-dot-org/assets/9256643/07466f9f-783a-4b98-aa12-a6e8a72270ae

## Tablet

https://github.com/code-dot-org/code-dot-org/assets/9256643/a7919044-2383-4232-8df1-b51239c04409

## Mobile

https://github.com/code-dot-org/code-dot-org/assets/9256643/0e740e14-a183-4a3d-8de8-4474f65144dd

## Keyboard navigation

https://github.com/code-dot-org/code-dot-org/assets/9256643/71d581b2-dc13-4bb7-be26-c899fae8db99

## Screen reader

https://github.com/code-dot-org/code-dot-org/assets/9256643/08c90d07-3faf-42c9-8719-361e031a4588

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/header.haml [hunk 1]
```diff
@@ -21,44 +21,17 @@
...
-          #sign_in_or_user
-            =view :sign_in_or_user
-          = view :help_button, hamburger_options
-          = view :hamburger, hamburger_options
-
-      #clear{:style=>'clear:both'}
-
-      #loc.desktop-feature
-        -if banner = @header['banner']
-          =view banner
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/header.haml:62]: I don't think code is relevant or used anymore.

File: pegasus/sites.v3/code.org/views/header.haml [hunk 2]
```diff
@@ -21,44 +21,17 @@
...
-
-#clear{:style=>'clear:both'}
+%nav.main
+  .left
+    %a{href: current_user ? CDO.studio_url("/home") : CDO.code_org_url}
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/header.haml:28]: I tested this locally, but would welcome extra testing if anyone has time!

Comment (TurnerRiley) [reply pegasus/sites.v3/code.org/views/header.haml:28]: I messed around with it locally and it looks good!

**Review by TurnerRiley — APPROVED** [review]
Fantastic!

**Pull request merged**